metricplugin/metric_export.go

```diff
@@ -143,13 +146,21 @@ func (mep *MetricExporterPlugin) Start(ipfsInstance *core.IpfsNode) error {
 		return errors.New("could not get Bitswap implementation")
 	}
 
+	// Get the bsnet.BitSwapNetwork implementation used in running Bitswap
+	// instance.
+	// This will break if the structure of the bs.Bitswap struct changes.
+	// Hopefully that doesn't happen too often...
+	// See https://stackoverflow.com/a/60598827
+	bsnetImpl, ok := (getUnexportedField(reflect.ValueOf(bitswapEngine).Elem().FieldByName("network"))).(bsnet.BitSwapNetwork)
```

Comment:
Do you know what the difference between the `Network` in ipfsNode and BitSwap is? As far as I understand the BitSwap network_impl.go implementation the BitSwap `Network` wraps an IPFS libp2p `Host`, including its `Network`. 
Put differently: Is this reflection actually necessary?

Reply:
Discussed offline :)